### DISCUSSION TITLE
What is the meaning and function of “nrrd” file? There is no explanation for the nrrd module.

### DISCUSSION THREAD

**Discussion by zhifeiji-han** [SCIInstitute/SCIRun#2298] [General] [answered]
I see nrrd file in scirun, but I don't know what it means and how to use it. The Module shown in the figure doesn't explain.

![image](https://user-images.githubusercontent.com/70640243/123767022-c0eac600-d8f9-11eb-9c8d-ff334a28a756.png)

**Comment by dcwhite** [comment]
Good question for @jessdtate

**Comment by tpat95** [comment, marked as answer]
The Nrrd format stands for "Nearly Raw Raster Data". It is both a program and a file format. SCIRun also uses the library within those modules. For example, the module ReadNrrd is a wrapper to loadNrrd() in the nrrd library.

About the program:
"Nrrd facilitates the pre-processing tasks that often precede visualization or segmentation (cropping, quantizing, resampling), as well as the basic analysis tasks which help to understand the structure of a dataset (histograms, maximum intensity projections, connected components). Nrrd has a persistent habit of replacing the throw-away one-time code that people often write to transform datasets, while also offering a convenient, flexible, and portable file format for storing the data."

About the file format:
" Like the nrrd library, the main virtue of the NRRD file format is stark simplicity. The NRRD header is simple ASCII text, one field per line. The fields in the header do not have a strict ordering, and most of them are optional. Most strings are case insensitive, and alternate forms of many of the identifiers and descriptors are allowed. When writing non-ASCII data, the byte ordering is recorded, but not altered to match one particular endianness. Key/value pairs (of strings) are stored in plain text, one pair per line.

The NRRD file format was also conceived as being somewhat analogous to the PPM format for color images: straight-forward, friendly to programmers, and descriptive of a sufficiently large class of data to be useful in research."

This describes the file format:
http://teem.sourceforge.net/nrrd/descformat.html

This is the main page about the nrrd program & file format:
http://teem.sourceforge.net/nrrd/index.html

**Comment by zhifeiji-han** [comment]
I see. Thank you for your reply！
